File: Reactivities-API/Reactivities.Domain/Activities/Filters/ActivityListFilters.cs
namespace Reactivities.Domain.Activities.Filters
{
    public class ActivityListFilters
    {
        public DateTime FromDate { get; set; } = DateTime.UtcNow;
        public DateTime ToDate { get; set; } = DateTime.MaxValue;
        public ActivityRelationship Relationship { get; set; }
    }
}

File: WebClient/src/app/models/activityListFilters.ts
import { makeAutoObservable } from 'mobx'
import ActivityRelationship from './activityRelationship'

export interface ActivityListFilters {
  fromDate: Date
  relationship: ActivityRelationship
}

export class ActivityListFilters implements ActivityListFilters {
  fromDate: Date = new Date()
  relationship: ActivityRelationship = ActivityRelationship.None

  constructor() {
    makeAutoObservable(this)
  }
}
